commit 302b93703ae9b9f2f78f4fd9e5d6fc7c1ec124fa
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2019-04-17T13:23:49-07:00 (6 months ago)
    
    TRITON-1415 sdc-image-sync should only put packages in one time.

diff --git a/tools/bin/sdc-image-sync b/tools/bin/sdc-image-sync
index 1a00c294..2eb0932e 100755
--- a/tools/bin/sdc-image-sync
+++ b/tools/bin/sdc-image-sync
@@ -45,22 +45,24 @@ function load_config_pkgs()
 
 function add_pkg()
 {
-	echo "add '$1' package for '$2' to the generic config file"
+	local pnm
+	pnm=$(sdc-papi --no-headers /packages/$1 | json name)
+	echo "add '$pnm' package for '$2' to the generic config file"
 
 	# Make sure we haven't already added it once before
-	egrep -s "^pkg_[0-9]*=${1}:" $usbmnt/config.inc/generic
+	egrep -s "^pkg_[0-9]*=${pnm}:" $usbmnt/config.inc/generic
 	if [ $? -ne 0 ]; then
-		out=$(sdc-papi --no-headers /packages?name=$pnm | \
-  json -a -d: name max_physical_memory max_swap quota cpu_cap max_lwps \
+		out=$(sdc-papi --no-headers /packages/$1 | \
+  json -d: name max_physical_memory max_swap quota cpu_cap max_lwps \
   zfs_io_priority uuid)
-  printf "pkg_%d=%s" $PKG_CNT $out >> $usbmnt/config.inc/generic
+		printf "pkg_%d=%s" $PKG_CNT $out >> $usbmnt/config.inc/generic
 
 	    # Just added an entry, so increment the cnt
 	    PKG_CNT=$((PKG_CNT + 1))
 	fi
 
 	# Change "role" pkg entry to use new pkg
-	nawk -F= -v entry="${2}_pkg" -v val=$1 '{
+	nawk -F= -v entry="${2}_pkg" -v val="$pnm" '{
 	    if ($1 == entry)
 		printf("%s=%s\n", $1, val)
 	    else
@@ -81,7 +83,7 @@ function update_sdc_zone_pkgs()
 	    -e 'this.smartdc_role=(this.tags.smartdc_role || "-")' \
 	    -e 'this.package_name=(this.package_name || "-")' \
 	    -c 'this.tags.smartdc_role && this.tags.smartdc_role!=="nat"' \
-	    -a smartdc_role package_name | \
+	    -a smartdc_role billing_id | sort -u | \
 	    while read role pkg
 	    do
 		# If this pkg is not in the config file, add it.
